### PR TITLE
defines project crd type

### LINKED ISSUE #967: Define Project type
Define a new CRD type `Project` that will have the following structure

```
type Project struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`
        
        Spec ProjectSpec `json:"spec"`
}
```

where `ProjectSpec` is:
```
type ProjectSpec struct {
	Name   string   `json:"name"`
}
```

we will take the `UID` from `metav1.ObjectMeta`

child of #966

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: defines project crd type

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #967 

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/crd/kubermatic/v1/project.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+// ProjectSpec is a specification of a project.
+type ProjectSpec struct {
+	Name string `json:"name"`
+}
+
+// ProjectStatus represents the current status of a project.
+type ProjectStatus struct {
+	Ready bool `json:"ready"`
```
Comment (p0lyn0mial) [inline api/pkg/crd/kubermatic/v1/project.go]: a project creation is a two-step process:

1. We need to create project 
2. We need to generate and assign proper `Role` and `RoleBindings`. 

The second step will be performed by a controller. I think we could also use `initializers` to achieve the same.

Comment (p0lyn0mial) [reply api/pkg/crd/kubermatic/v1/project.go]: I think this is also true for other types like `cluster`, so defining `Ready` field for each project resource seems to be a bit redundant.

**Review by mrIncompetent — DISMISSED** [review]

**Comment by p0lyn0mial** [comment]
fails on code generation, i'm putting this pull on hold until I'll prepare a pull that regenerates the clientset.

**Comment by p0lyn0mial** [comment]
@mrIncompetent please re-tag the pull.

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**